File: Day_4/118. Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascal;
        for(int i=0; i<numRows; i++){
            vector<int>newRow(i+1, 1);
            for(int j=1; j<i; j++){
                newRow[j] = pascal[i-1][j] + pascal[i-1][j-1];
            }
            pascal.push_back(newRow);
        }
        return pascal;
    }
};

File: Day_4/118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascal = []
        for i in range(1, numRows+1):
            newRow = [1]*i
            for j in range(1, i-1):
                newRow[j] = pascal[i-2][j] + pascal[i-2][j-1]
            pascal.append(newRow)
        return pascal
